Welcome to version 2.0

karllindmark opened the pull request:
Battle Chat is being bumped to version 2.0

Changes include, but are not limited to:
- Gradle support!!!
- Navigation Drawer (huzzah!)
- Fragments
- Loaders
- Better looking UI (thanks @humphreybc!)
- Gson!

Commit:
Adding the library to the module

Commit:
Chat is now working via the SlidingMenu.

Content is to be displayed prior to the user selecting a contact to talk to (or we just default to the saved ID or something).

More thorough testing required.

Commit:
Updated the tests to include the GRAVATAR URL.

Q: Should we keep store the hash, or the url? URL is easier to use, as users without a gravatar don't have a hash.

We also need to consider recycling avatars in some way (but I reckon that ImageLoader might be able to handle things like that).

Commit:
Porting the test fix from previous branch (missed out the merge into dev)

peter-budo committed:
Managing local dependency of sliding menu since not available on Sonatype

Commit:
Merge pull request #49 from peter-budo/slidingmenu-impl

Build dependencies management

Commit:
Addressing the data retrieval issues in the login process caused by Battlelog feeding mobile devices a stripped desktop version

Commit:
Adding further permissions for the app to be able to store files

Commit:
Adding the account box to the SlidingMenu

Commit:
Adding ImageLibrary dependency to the pom's

Commit:
Merging the dependencies

Commit:
Adding the holo text appearance to TextViews

Commit:
Updating the minSdkVersion to 14 (4.0) instead of 2.2 (8)

Commit:
Making sure the appropriate indicator is being shown (on newer API's as well)

Commit:
Addressing bug where HTML entities were present in the content. Github issue #47.

Commit:
Addressing inter-fragment/activity communication and state saving

The fragments should now be functioning as expected to, and the state saving as well. Hopefully. ;-)

Commit:
Adding check to see that the ListView exists before trying to use it

Commit:
Making sure we can't accidently try to access the UI while it ain't there

Commit:
[WIP] Stash changes to PresenceType

Commit:
Battle Chat is now getting closer to 2.0

Commit:
Add underlying support for groups, not sure how to make it useful though

Commit:
Update .gitignore

Commit:
Remove state handling as it causes crashes - state works good enough "out of the box"

Commit:
Overall refactoring and fixing

Commit:
Remove obsolete TODO

Commit:
Remove obsolete debug outputs

Commit:
Move base functionality to parent class

Commit:
Update selector for the list items

Commit:
Close chats on back pressed

Commit:
Address issue where notification was not update onResume()

Commit:
Correct issue where notification was shown all the time

Commit:
Make sure we don't overlap fragmets on rotation

Commit:
Add TODO to project ;-)

Commit:
Update MainActivity to handle rotation correctly

Commit:
Refactor StartupFragment

Commit:
Adjust startup fragment layout

Commit:
Login activity is now completed (for the time being)

Commit:
Minor fixes

Commit:
Extract notification functionality to NotificationHelper

Commit:
Fix a bunch of Android lint issues

Commit:
Further complying to lint

Commit:
Adjust login to include error message a bit better

Commit:
Address issues with service (keeping references)

Commit:
Start work on About

Commit:
Finish author information fragment

Commit:
Add collaborators fragment

Commit:
Add open source fragment

Commit:
Made text on about app fragment nicer

Commit:
Add license information to all xml files

karllindmark commented:
@peter-budo

Commit:
Remove hardcoded email

Commit:
Remove TODO comment from code

karllindmark commented:
@gustavkarlsson

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/BattleChat.java:
```diff
@@ -0,0 +1,63 @@
+import com.ninetwozero.battlechat.misc.Keys;
+
+public class BattleChat extends Application {
+    public final static String TAG = "com.ninetwozero.battlechat";
```
Obsolete `getSimpleName()` is better

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/base/asynctasks/BaseLogoutTask.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    @Override
+    protected Void doInBackground(Void... params) {
```
params not used

karllindmark replied:
Params not used, but has to be there due to how `doInBackground(...)` is being defined. `Void...` corresponds to blank parameters, in this case.

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/base/ui/BaseListAdapter.java:
```diff
@@ -0,0 +1,98 @@
+    }
+
+    final public void setText(final View container, final int resourceId, final String text) {
+        setText(container, resourceId, text, -1);
```
`-1` magic number

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/base/ui/BaseListFragment.java:
```diff
@@ -0,0 +1,44 @@
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+        this.layoutInflater = inflater;
```
Is that assignment necessary? Also returning null?

karllindmark replied:
Yeah, it's to keep the instance for further use. :-)

Returning null won't be an issue in this case, as we don't use the return value. However, `onCreateView(LayoutInflater, ViewGroup, Bundle)` requires a view to be returned, but that's handled by the "kids".

karllindmark replied:
`Return the View for the fragment's UI, or null.` (from the docs for android.support.v4.app.ListFragment)

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/network/IntelLoader.java:
```diff
@@ -0,0 +1,67 @@
+import com.ninetwozero.battlechat.factories.GsonFactory;
+import com.ninetwozero.battlechat.network.exception.Failure;
+
+public class IntelLoader<T> extends AsyncTaskLoader<Result> {
```
Thief! not even changing class name ROFL

karllindmark replied:
I'm quite transparent with what I've been copying between the projects - I'd like to see some sort of "Common Battlelog library" in the future if possible. For things like this, for example. :-D

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/datatypes/NoResultExpected.java:
```diff
@@ -0,0 +1,21 @@
+
+package com.ninetwozero.battlechat.datatypes;
+
+public class NoResultExpected {
```
Weird class name. Am I suppose to understand that loader should not return any result and throw exception if there is some result?

karllindmark replied:
It's supposed to be some sort of "fire & forget", but the loaders themselves want us to pass a "datatype" for the generics. I suppose I could use `Void` instead, but not sure if it's better (from a readability point of view).

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/datatypes/Session.java:
```diff
@@ -0,0 +1,129 @@
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+
+import com.ninetwozero.battlechat.misc.Keys;
```
Cry of despair! `misc`

karllindmark replied:
Moving them to root package ;-)

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/datatypes/Session.java:
```diff
@@ -0,0 +1,129 @@
+
+        editor.remove(Keys.Session.USER_ID);
+        editor.remove(Keys.Session.USERNAME);
+        // editor.remove(Keys.Session.EMAIL);
```
Remove any commented out code

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/factories/UrlFactory.java:
```diff
@@ -0,0 +1,67 @@
+    }
+
+    public static URL buildOpenChatURL(final String userId) {
+        return createURL(String.format("comcenter/getChatId/%s/", userId));
```
I realize we are not doing it in other project too. Any stream or string manipulation should include encoding `UTF-8`

karllindmark replied:
Do you have an example on how this usage would be? :cake:

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/interfaces/AboutInfoRow.java:
```diff
@@ -0,0 +1,56 @@
+        this.title = title;
+        this.type = Type.HEADER;
+    }
+
```
Extra line :tongue:

karllindmark replied:
Fixed!

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/network/Result.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    @Override
+    public String toString() {
```
Unnecessary polution

peter-budo commented on app/src/main/java/com/ninetwozero/battlechat/services/BattleChatService.java:
```diff
@@ -0,0 +1,312 @@
+                final List<Chat> unreadChats = findUnreadChats(comCenter.getInformation().getChats());
+
+                Log.d(TAG, "Number of unread chats: " + unreadChats.size());
+                for (Chat chat : unreadChats) {
```
Ugly for loop inside for loop

karllindmark replied:
Fixed!

Commit:
Convert magic number into constant

Commit:
Move Keys from ./misc --> ./

Commit:
Minor refactorings per PR comments

Commit:
Create nicer 'changelog' fragment layout

Commit:
Add card layout to changelog

Commit:
Update build tools & co

Commit:
Add BugSense logging to the app

Add BugSense logging to the app. Also, optimize the imports.

Commit:
Update navigation drawer selector colors

Commit:
Make UI more responsive (login + about)

Commit:
Make chat tablet friendly

karllindmark commented:
@peter-budo: Could you try this on your Nexus 7 and post a screenshot of 1) the login screen and 2) the "main" activity? That would be helpful for our future adventures. :+1:

Commit:
Update about to match BF4 Intel

Commit:
[WIP] Porting to Volley

Commit:
Update build configuration

Commit:
Volley now works as expected - woo

Commit:
Minor UI touch-ups

Commit:
[WIP]

Commit:
[WIP] Delegate work to IntentServices

Setting up work towards IntentServices where the UI will talk to the DB
instead of the website itself.

Commit:
Cache messages in awesomeness

Commit:
Cache friends in awesomeness too

Commit:
BattleChat should now ready for release 2.0
